upup/pkg/fi/utils/yaml.go

```diff
@@ -28,6 +28,8 @@ func YAMLToJSON(yamlBytes []byte) ([]byte, error) {
 }
 
 // YamlUnmarshal unmarshals the yaml content to an interface
+// Deprecated: normally you want something like kopscodecs.Decode,
+// so that we can convert apiVersions.
```

Comment:
One doesn't need to and can't convert apiVersions for three uses of this: BootConfig, NodupConfig, and the `--values` flag of `kops toolbox template`. So I don't think it's appropriate to deprecate.

Reply:
Yes, that's fair.  Changing to a note.